src/main/flight/mixer.c

```diff
@@ -568,7 +568,9 @@ void calculateThrottleAndCurrentMotorEndpoints(void)
             rcThrottlePrevious = rcCommand[THROTTLE];
             throttle = rcCommand[THROTTLE] - rcCommand3dDeadBandHigh;
             currentThrottleInputRange = rcCommandThrottleRange3dHigh;
-        } else if((rcThrottlePrevious <= rcCommand3dDeadBandLow)) {
+        } else if((rcThrottlePrevious <= rcCommand3dDeadBandLow &&
+                !isModeActivationConditionPresent(BOX3DONASWITCH)) || 
```

Comment:
Shouldn't these conditions be added for the `INVERTED` case above as well?

Reply:
It doesn't do any dshot reversing because that won't work in flight (at least not with blheli_s currently).   But should work on both dshot and PWM (I'll test that out when it isn't raining out) Need to set up 3d mode on the ESC.

Reply:
The above issue is just to fix so it is doing the right thing when in deadband.  I won't want to rely on the previous throttle and only use what the switch position is.

Reply:
Hehe, confused myself there. I wasn't aware that `isMotorsReversed()` was always false, and the block you removed below was just removing dead code.

Reply:
Good point. I wonder how much unwanted delay a disarm ESC / switch direction / arm ESC would add, if it was done in-flight. (The risk of this failing and the craft ending up with a subset of its motors spinning is another factor to be taken into consideration.)

Reply:
The best way would be to modify the ESC code to allow the dshot commands while active, but that might be a can of worms...   

I thought you mentioned KISS escs might already allow this when you were doing work with them?

Reply:
I know the KISS ESCs allow beeping while armed (or at least they did at some stage), and I see this as more of a bug than a feature, since it's obviously affecting the motor's spin quite badly. In this respect, I am not sure if allowing commands while armed is the right way to go. The only candidates with some justification would be the forward / reverse commands, and even there, it only really makes sense in the case where the motor speed is near idle. Not requiring the ESC to be disarmed removes one layer of protection against accidental command execution, while not completely eliminating the potential for not all ESCs picking up the command to change.

Maybe we should first look into adding command feedback to ESC telemetry, and then cross the bridge of in-flight ESC commands?

Reply:
Yeah sounds good to me.